Taskfile.yaml

```
@@ -0,0 +1,71 @@
+version: "3"
+
+vars:
+  VENV_DIR: .venv
+  REPO_URL: https://github.com/EBI-Metagenomics/nf-modules
+
+tasks:
+  # Setup Python virtual environment using uv
+  setup-venv:
+    desc: "Bootstrap Python virtual environment with uv"
+    cmds:
+      - uv venv {{.VENV_DIR}}
+      - uv pip install nf-core pre-commit
+    status:
+      - test -d {{.VENV_DIR}}
+    silent: true
+
+  # Module management tasks
+  module-create:
+    desc: "Create a new nf-core module interactively"
+    deps: [setup-venv]
+    cmds:
+      - "uv run nf-core modules -g {{.REPO_URL}} create"
+    interactive: true
+
+  modules-lint:
+    desc: "Lint a specific module (usage: task modules-lint -- <module_name>)"
+    deps: [setup-venv]
+    cmds:
+      - uv run nf-core modules -g {{.REPO_URL}} lint {{.CLI_ARGS}}
+    interactive: true
+
+  # Subworkflow management tasks
+  subworkflow-create:
+    desc: "Create a new nf-core subworkflow interactively"
+    deps: [setup-venv]
+    cmds:
+      - "uv run nf-core subworkflows -g {{.REPO_URL}} create"
+    interactive: true
+
+  subworkflow-lint:
+    desc: "Lint a specific subworkflow (usage: task subworkflow-lint -- <subworkflow_name>)"
+    deps: [setup-venv]
+    cmds:
+      - uv run nf-core subworkflows -g {{.REPO_URL}} lint {{.CLI_ARGS}}
+    interactive: true
+
+  # Pre-commit tasks
+  pre-commit:
+    desc: "Run pre-commit hooks on staged files"
+    deps: [setup-venv]
+    cmds:
+      - "uv run pre-commit run"
+
+  pre-commit-all:
+    desc: "Run pre-commit hooks on all files"
+    deps: [setup-venv]
+    cmds:
+      - "uv run pre-commit run --all-files"
+
+  pre-commit-install:
+    desc: "Install pre-commit hooks"
+    deps: [setup-venv]
+    cmds:
+      - "uv run pre-commit install"
+
+  # Utility tasks
+  clean:
+    desc: "Clean up virtual environment"
+    cmds:
+      - rm -rf {{.VENV_DIR}}
```

add `prompt` for destructive tasks

```suggestion
  clean:
    desc: "Clean up virtual environment"
    prompt: "This will destroy your virtual env, okay?"
    cmds:
      - rm -rf {{.VENV_DIR}}
```